cranelift/codegen/src/isa/aarch64/inst.isle

```diff
@@ -1997,3 +2015,68 @@
         (value_regs
           (alu_rrr op ty x_lo y_lo)
           (alu_rrr op ty x_hi y_hi))))
+
+;; Float vector compare helpers ;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;
+
+(decl zero_value (Imm64) Imm64)
+(extern extractor zero_value zero_value)
+
+(decl zero_value_f32 (Ieee32) Ieee32)
+(extern extractor zero_value_f32 zero_value_f32)
+
+(decl zero_value_f64 (Ieee64) Ieee64)
+(extern extractor zero_value_f64 zero_value_f64)
+
+(decl float_cc_cmp_zero_to_vec_misc_op (FloatCC) VecMisc2)
+(extern constructor float_cc_cmp_zero_to_vec_misc_op float_cc_cmp_zero_to_vec_misc_op)
+
+(decl float_cc_cmp_zero_to_vec_misc_op_swap (FloatCC) VecMisc2)
+(extern constructor float_cc_cmp_zero_to_vec_misc_op_swap float_cc_cmp_zero_to_vec_misc_op_swap)
+
+;; Match valid generic compare to zero cases
+(decl fcmp_cond_zero (FloatCC) FloatCC)
+(extern extractor fcmp_cond_zero fcmp_cond_zero)
+
+;; Match not equal compare to zero separately as it requires two output instructions
+(decl fcmp_cond_not_eq (FloatCC) FloatCC)
+(extern extractor fcmp_cond_not_eq fcmp_cond_not_eq)
+
+(decl f_comp_inst (FloatCC Reg VectorSize) Reg)
```

Comment:
Sorry, I should've said "and similar below" to earlier comments about naming the above constructors. Could you make all of these added ones more descriptive as well? E.g. `f_comp_inst` doesn't give me any hint that this might have to do with zero comparisons; at best I might buess that it means "create a floating point comparison instruction", which sounds like a generic constructor.

Doc comments on each (one line is fine) to say what they do would also be helpful. Thanks!